src/lab_orchestrator_lib_auth/auth.py

```diff
@@ -55,29 +58,28 @@ def decode_auth_token(token: str, secret_key: str) -> Optional[LabInstanceTokenP
     :param secret_key: Key that should be used to decrypt the token.
     :return: The data that is contained in the token or None if decode goes wrong.
     """
-    try:
-        data = jwt.decode(token, secret_key, algorithms=['HS256'])
-        return LabInstanceTokenParams(
-            lab_id=data['lab_instance']['lab_id'],
-            lab_instance_id=data['lab_instance']['lab_instance_id'],
-            namespace_name=data['lab_instance']['namespace_name'],
-            allowed_vmi_names=data['lab_instance']['allowed_vmi_names']
-        )
-    except:
-        return None
-
-
-def verify_auth_token(token: str, vmi_name: str, secret_key: str) -> Optional[LabInstanceTokenParams]:
+    data = jwt.decode(token, secret_key, algorithms=['HS256'])
+    return LabInstanceTokenParams(
+        lab_id=data['lab_instance']['lab_id'],
+        lab_instance_id=data['lab_instance']['lab_instance_id'],
+        namespace_name=data['lab_instance']['namespace_name'],
+        allowed_vmi_names=data['lab_instance']['allowed_vmi_names']
+    )
+
+
+
+def verify_auth_token(token: str, vmi_name: str, secret_key: str) -> Tuple[bool, LabInstanceTokenParams]:
     """Decodes a token and verifies if it's valid.
 
     Checks if the vmi_name is allowed in the token.
 
     :param token: The token to decode and verify.
     :param vmi_name: The vmi_name the user wants to use.
     :param secret_key: Key that is used to decrypt the token.
-    :return: The data that is contained in the token or None if the the token or vmi name was invalid or not allowed.
+    :return: The result of the verification as a boolean and the data contained in the token
     """
+
     data = decode_auth_token(token, secret_key)
-    if data is None or vmi_name not in data.allowed_vmi_names:
-        return None
-    return data
+    is_forbidden = data is None or vmi_name not in data.allowed_vmi_names
```

Comment:
the `verify_auth_token` method is used to shortly check if the authentication was successful, and because you have changed the `decode_auth_token` method, you need to catch the errors that can appear on wrong tokens instead of checking if the result is none: https://pyjwt.readthedocs.io/en/latest/api.html?highlight=decode#exceptions

Reply:
I am not sure if you would want to catch all errors here and just return false or rather pass through the errors? Passing them through would give more possibilites for error handling which leads to more useful error messages for the end user.

Reply:
ok, but then add documentation for the exceptions here too.

Reply:
done